richtextfx/src/main/java/org/fxmisc/richtext/model/SegmentOps.java

```diff
@@ -68,15 +68,19 @@ public String getText(Either<L, R> seg) {
     }
 
     @Override
-    public Either<L, R> subSequence(Either<L, R> seg, int start, int end) {
-        return seg.map(l -> lOps.subSequence(l, start, end),
-                       r -> rOps.subSequence(r, start, end));
+    public Optional<Either<L, R>> subSequence(Either<L, R> seg, int start, int end) {
+        return seg.unify(ll -> seg.unify(l -> lOps.subSequence(l, start, end).map(Either::left),
+                                         e -> Optional.empty()),
+                         rr -> seg.unify(e -> Optional.empty(),
+                                         r -> rOps.subSequence(r, start, end).map(Either::right)));
```

Comment:
Unlike the `join` method, you are calling `unify` twice on the same value (`seg`). Therefore, the `e -> Optional.empty()` is never executed, because we already know it is the other case.

Seems like type inference cannot figure out the types. Try replacing `Either::left` with `Either::<L, R>left`.